frontend/src/SignUp.jsx

```diff
@@ -39,98 +69,143 @@ const SignUp = () => {
+        <header>
+          <div className="text-center m-15 text-2xl font-bold text-indigo-50 text-shadow-xl ">
+            <h1 className="drop-shadow-[0px_0px_39px_rgba(247,247,247,.7)]">
+              Alpha-Edge
+            </h1>
+          </div>
+        </header>
+        <form className="flex flex-col bg-indigo-50 p-8 rounded-md shadow-xl/40 w-150 m-auto">
+          <h3 className="font-bold text-3xl p-1 text-center">Sign Up</h3>
+          <p className="font-bold text-sm p-0 text-center">
+            Please sign up to continue{" "}
+          </p>
+          <InputBox
+            placeholder="John Doe"
+            label="Full Name"
+            name="name"
+            value={formData.name}
+            handleFormChange={handleFormChange}
+          />
+
+          <InputBox
+            placeholder="johndoe@gmail.com"
+            label="Email"
+            name="email"
+            value={formData.email}
+            handleFormChange={handleFormChange}
+          />
+          <InputBox
+            placeholder="Password"
+            label="Password"
+            name="password"
+            value={formData.password}
+            handleFormChange={handleFormChange}
+          />
+          <InputBox
+            placeholder="Username"
+            label="Username"
+            name="username"
+            value={formData.username}
+            handleFormChange={handleFormChange}
+          />
+          <button
+            type="submit"
+            id="signup"
+            className="m-2 bg-green-400 shadow-xl/10 shadow-slate-900  hover:brightness-110"
+            onClick={handleNext}
+          >
+            Next
+          </button>
+          {submitResult != null && (
+            <p className="text-center font-bold ">{submitResult}</p>
+          )}
+
+          <button
+            onClick={() => {
+              navigate("/");
+            }}
+            id="login"
+            className="m-2 bg-fuchsia-950 text-white shadow-xl/10 shadow-slate-900 hover:brightness-110"
+          >
+            Back to login
+          </button>
+        </form>
+      </div>
+    );
+  } else {
+    return (
+      <div className="flex flex-col">
+        <header>
+          <div className="text-center m-15 text-2xl font-bold text-indigo-50 text-shadow-xl ">
+            <h1 className="drop-shadow-[0px_0px_39px_rgba(247,247,247,.7)]">
+              Alpha-Edge
+            </h1>
+          </div>
+        </header>
+        <div className="flex flex-col bg-indigo-50 p-8 rounded-md shadow-xl/40 w-150 m-auto">
+          <h2 className="self-center font-bold text-xl">
+            Thank You For Joining Alpha-Edge
+          </h2>
+          <h3 className="self-center font-med text-md">
+            To complete your registration, continue with the steps below:
+          </h3>
+          <h3 className="self-center font-bold text-xl mt-7">
+            Choose 3 sectors of interest
+          </h3>
+          <SectorList />
```

Comment:
I actually think that you might not need to create a context (over-engineering) and use a callback instead.

In this file, add:
```
const [sectorsSelected, setSectorsSelected] = useState([]);
<SectorList sectorsSelected={sectorsSelected} setSectorsSelected={setSectorsSelected} />
```

and you can just remove the `useState` in both `SectorList` component and `SignUpInfoContext`

```

I didn't see you have passed `callback` as a prop before in your entire internship. Could you share an instance where you have try to used them?